### PR TITLE
cs fixes

### PR DESCRIPTION
**Pull request by HeahDude**

### CODE REVIEW

**Comment by HeahDude** [comment]
Please, consider adding those changes in your work or discuss them.

No needs to merge :)

File: src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/RequestResolver.php [hunk 1]
```diff
@@ -27,7 +27,7 @@
      */
     public function supports(Request $request, ArgumentMetadata $argument)
     {
-        return $argument->getType() === Request::class || is_subclass_of($request, $argument->getType());
+        return $argument->getType() === Request::class || is_subclass_of(Request::class, $argument->getType());
```
Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/RequestResolver.php]: This has a huge perf gain:
```
<?php

abstract class AbstractTest
{
	public function yieldClass()
	{
		return get_class($this).' | '.$this->getClass();
	}

	abstract protected function getClass();
}

class Test extends AbstractTest
{
	public function test() {}
	public function test2() {}
	public function test3() {}
	
	protected function getClass()
	{
		return __CLASS__;
	}
}

$a = new Test();

echo is_subclass_of($a, 'AbstractTest') ? 'true' : 'false';
// echo is_subclass_of('Test', 'AbstractTest');
```

see https://blackfire.io/profiles/compare/cb9993ca-4cf4-410c-9e73-b6fd1d950dd8/graph

File: src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php [hunk 2]
```diff
@@ -59,7 +59,7 @@ private function isVariadic(\ReflectionParameter $parameter)
      *
      * @return bool
      */
-    private function hasDefaulValue(\ReflectionParameter $parameter)
+    private function hasDefaultValue(\ReflectionParameter $parameter)
```
Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php]: Others may be trivial, but this one is needed :)

Comment (iltar) [inline src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php]: Oh wow, good catch! I must have misread this a hundred times

Comment (HeahDude) [inline src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php]: That's what happen when you're the writer, plus using auto-completion :)

File: src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/RequestResolver.php (hunk 1 above)
Comment (iltar) [inline src/Symfony/Component/HttpKernel/Controller/ArgumentValueResolver/RequestResolver.php]: Hm that's actually really nice, I skimmed the methods and it stated `$object` hence I passed the object, unaware of this difference. Luckily this case barely ever gets hit but nice catch nevertheless

File: src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php (hunk 2 above)
Comment (iltar) [inline src/Symfony/Component/HttpKernel/ControllerMetadata/ArgumentMetadataFactory.php]: \*cough* never happened to me before \*cough*

**Comment by iltar** [comment]
I will merge this one and add a test for the Request fix you proposed, if you want to can add the fix yourself, feel free to make a new PR, I don't want to take your credit ;)

File: src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolverTest.php [hunk 3]
```diff
@@ -33,63 +36,108 @@ public function testGetArguments()
             new DefaultArgumentValueResolver(),
         );
 
-        $resolver = new ArgumentResolver($factory, $argumentValueResolvers);
+        self::$resolver = new ArgumentResolver($factory, $argumentValueResolvers);
+    }
```
Comment (iltar) [inline src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolverTest.php]: I assume this is due to the statelessness and no mocks? Looks good

**Pull request merged**